3.analysis/scripts/3.t_tests.py

```diff
@@ -470,3 +474,89 @@ def generate_latent_df(latent_df, category_ids):
+    other_latent_df = latent_df[latent_df['ModelID'].isin(other_ids)]
+    other_latent_float_df = other_latent_df.drop(columns=["ModelID"]).reset_index(drop=True)
+
+    # Perform t-tests comparing current disease vs the rest for each latent dimension
+    t_test_results = ttest_ind(disease_latent_float_df, other_latent_float_df)
+    t_test_results_df = pd.DataFrame(t_test_results).T
+    t_test_results_df.columns = ["t_stat", "p_value"]
+    t_test_results_df['comparison'] = f"{disease} vs Rest"
+    t_test_results_df['latent_feature'] = t_test_results_df.index + 1
+    t_test_results_df['z_dim'] = 'z_' + t_test_results_df['latent_feature'].astype(str)
+    t_test_results_df['group'] = t_test_results_df['t_stat'].apply(lambda x: disease if x > 0 else 'Rest')
+
+    # Filter significant latent features
+    significant_latent_features = t_test_results_df[t_test_results_df['p_value'] < 0.05]
+
+    # Loop through all unique pathways (Terms) in the GSEA DataFrame
+    for pathway in gsea_results_df['Term'].unique():
+        # Filter for the pathway
+        filtered_gsea_df = gsea_results_df[gsea_results_df['Term'] == pathway]
+        
+        # Further filter by significant latent features
+        filtered_gsea_df = filtered_gsea_df[filtered_gsea_df['z_dim'].isin(significant_latent_features['z_dim'])]
+        
+        # Merge GSEA DataFrame with t-test DataFrame to get group information
+        merged_df = pd.merge(filtered_gsea_df, significant_latent_features[['z_dim', 'group']], on='z_dim', how='inner')
+
+        # Group by 'group' and collect ES values
+        grouped_data = merged_df.groupby('group')['es'].apply(list)
+
+        # Ensure we have data for both groups
+        if len(grouped_data) == 2 and all(len(vals) > 1 for vals in grouped_data):
+            # Perform ANOVA
+            f_statistic, p_value = f_oneway(*grouped_data)
+
+            # Determine which group has higher enrichment score
+            disease_mean = np.mean(grouped_data[disease])
+            rest_mean = np.mean(grouped_data['Rest'])
+            higher_group = disease if disease_mean > rest_mean else 'Other Types'
+
+            # Store results
+            anova_results.append({
+                'Cancer Type': disease,
+                'Pathway': pathway,
+                'F-statistic': f_statistic,
+                'p-value': p_value,
+                'Higher in': higher_group
+            })
+
+# Convert the results to a DataFrame
+anova_results_df = pd.DataFrame(anova_results)
+
+# Apply a significance threshold (e.g., p < 0.05)
+significant_anova_results_df = anova_results_df[anova_results_df['p-value'] < 0.05]
+
+# Save the results to a CSV file
+anova_dir = pathlib.Path("./results/cancer_type_anova_results.csv")
+significant_anova_results_df.to_csv(anova_dir, index=False)
+
+# Display significant pathways
+significant_anova_results_df.sort_values(by='F-statistic', key=abs, ascending=False).head(50)
```

Comment:
If a row has other types, the given type is significantly lower than everything else. Because each cancer is reviewed, it is easy enough to sort through and find which are the most significant in that contribution, especially if you sort the df by pathway, but there might be a good way to visually display this too!